swing-lib/src/main/java/org/triplea/swing/KeyTypeValidator.java

```diff
@@ -21,31 +21,31 @@ public void attachKeyTypeValidator(
       final JTextComponent textComponent,
       final Predicate<String> dataValidation,
       final Consumer<Boolean> action) {
-    DocumentListenerBuilder.attachDocumentListener(
-        textComponent,
-        () ->
-            new Thread(
-                    () -> {
-                      if (validationIsInFlight.compareAndSet(false, true)) {
-                        // sleep to delay our current check and allow
-                        // the check to be done once at the end of any
-                        // further input that might still be in-flight.
-                        if (!Interruptibles.sleep(200)) {
-                          validationIsInFlight.set(false);
-                          return;
-                        }
-                        // release the boolean lock, if we get another validation
-                        // request, allow it to enter queue. This way even if 'getText()'
-                        // returns us stale data, that new request will have a chance to
-                        // operate on the most recent.
-                        validationIsInFlight.set(false);
+    new DocumentListenerBuilder(
```

Comment:
Most of the updates here are white space changes and line re-ordering. The update here is only using the new `DocumentListenerBuilder` API.